clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2196,8 +2197,32 @@ llvm::ConstantInt *CodeGenModule::CreateKCFITypeId(QualType T) {
   if (getCodeGenOpts().SanitizeCfiICallNormalizeIntegers)
     Out << ".normalized";
 
-  return llvm::ConstantInt::get(Int32Ty,
-                                static_cast<uint32_t>(llvm::xxHash64(OutName)));
+  uint32_t OutHash = static_cast<uint32_t>(llvm::xxHash64(OutName));
+  const auto &Triple = getTarget().getTriple();
+  if (Triple.isX86() && Triple.isArch64Bit() && Triple.isOSLinux()) {
```

Comment:
@sirmc Thank you for the suggestion. I looked at the code and as far as I can tell, `CreateKCFITypeId()` is only called here:  https://github.com/llvm/llvm-project/blob/b89e774672678ef26baf8f94c616f43551d29428/clang/lib/CodeGen/CodeGenModule.cpp#L2840, which seems to guarantee that `T` will always be a function type. Regardless, I think your suggestion is reasonable, in case `CreateKCFITypeID()` is called elsewhere in the future, so I incorporated this additional check into the PR.